### PR TITLE
Add `Sync: Jira` label

### PR DESCRIPTION
**Pull request by chadwhitacre**
This PR adds a `Sync: Jira` label to support an automation to create Jira shadow tickets for certain GitHub Issues (https://github.com/getsentry/team-ospo/issues/63).

### CODE REVIEW

File: .github/labels.yml [hunk 1]
```diff
@@ -15,6 +15,11 @@
   color: '3E5E06'
   description: 'add to a merged PR to revert it (skips CI)'
 
+# Unito - notion.so/52d10dbfad474328850319e48b057a5b
+- name: "Sync: Jira"
```
Comment (evanpurkhiser) [inline .github/labels.yml:19]: Is it truly a sync, or does it just copy?

**Review by evanpurkhiser — APPROVED** [review]

File: .github/labels.yml (hunk 1 above)
Comment (chadwhitacre) [reply .github/labels.yml:19]: [RTFM](https://www.notion.so/GitHub-Jira-Integration-52d10dbfad474328850319e48b057a5b) ;-)

tl;dr It's a sync.

**Comment by chadwhitacre** [comment]
> Added: the "Team: Dev Infrastructure" label in the repo is not expected. It will be deleted.

Not ideal.
